classes/task/fetch_recording_metadata_task.php

```diff
@@ -0,0 +1,145 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_bigbluebuttonbn\task;
+
+defined('MOODLE_INTERNAL') || die();
+
+// Time between checking the same record.
+const TIME_BETWEEN_SUBSEQUENT_CHECKS = 20 * MINSECS;
+// After 2 weeks from the meeting start date, if no recording metadata is
+// returned, consider the meeting as recording-free.
+const TIME_BEFORE_FLAGGING_AS_NOT_RECORDED = 2 * WEEKSECS;
+// This is a limit documented in code @  bigbluebuttonbn_get_recordings_array_fetch_page function call.
+const BBB_API_GET_RECORDINGS_RECORDS_PER_REQUEST = 25;
+// Default processing limit for backlog of BBB sessions.
+const DEFAULT_PROCESSING_LIMIT = 100;
+
+/**
+ * A scheduled task to fetch the recordings' metadata and store it for further reporting
+ *
+ * @package    mod_bigbluebuttonbn
+ * @author     Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright  Catalyst IT, 2021
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class fetch_recording_metadata_task extends \core\task\scheduled_task {
+
+    /**
+     * Get a descriptive name for this task (shown to admins).
+     *
+     * @return string
+     */
+    public function get_name() {
+        return get_string('fetchmetaforrecordings', 'mod_bigbluebuttonbn');
+    }
+
+    /**
+     * Run chat cron.
+     */
+    public function execute() {
+        global $CFG, $DB;
+        require_once(__DIR__.'/../../locallib.php');
+
+        $sql = "SELECT recordid, id, *
+                  FROM {bigbluebuttonbn_logs}
+                 WHERE recordid IS NOT NULL
+                   AND log = :log
+                   AND meta LIKE :recordtrue
```

Comment:
These should move over to $DB->sql_like, which handles them properly for cross-db support